File: 2003/CS101SamplesAll/CS101SamplesVSTO/WorkingWithTheActionsPane/WorkingWithTheActionsPane/ThisDocument.cs
using System;
using System.Data;
using System.Drawing;
using System.Windows.Forms;
using Microsoft.VisualStudio.Tools.Applications.Runtime;
using Word = Microsoft.Office.Interop.Word;
using Office = Microsoft.Office.Core;

namespace WorkingWithTheActionsPane
{
    public partial class ThisDocument
    {
        ProductLookupUserControl uc;
        Button b1, b2;

        private void ThisDocument_Startup(object sender, System.EventArgs e)
        {
            // Create a new instance of the user control
            uc = new ProductLookupUserControl();
            
            // Create a test button
            b1 = new Button();
            b1.Text = "Button 1";
            b1.Click += new EventHandler(b1_Click);

            // Create another test button
            b2 = new Button();
            b2.Text = "Button 2";
            b2.Click += new EventHandler(b2_Click);

            // Add all three controls to the ActionsPane, show it, then
            // add an event handler to fire if the orientation of the
            // ActionsPane changes.
            this.ActionsPane.Controls.AddRange(new Control[] { uc, b1, b2 });
            this.ActionsPane.Show();
            this.ActionsPane.OrientationChanged += new EventHandler(ActionsPane_OrientationChanged);
        }

        void b1_Click(object sender, EventArgs e)
        {
            MessageBox.Show("You clicked button 1");
        }

        void b2_Click(object sender, EventArgs e)
        {
            MessageBox.Show("You clicked button 2");
        }

        /// <summary>
        /// If the toolbar is docked from side to top or bottom, the controls can
        /// be laid out left-to-right instead of top-to-bottom.
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="e"></param>
        void ActionsPane_OrientationChanged(object sender, EventArgs e)
        {
            if (this.ActionsPane.Orientation == Orientation.Vertical)
            {
                this.ActionsPane.StackOrder = Microsoft.Office.Tools.StackStyle.FromTop;
            }
            else
            {
                this.ActionsPane.StackOrder = Microsoft.Office.Tools.StackStyle.FromLeft;
            }
        }

        private void ThisDocument_Shutdown(object sender, System.EventArgs e)
        {
        }

        #region VSTO Designer generated code

        /// <summary>
        /// Required method for Designer support - do not modify
        /// the contents of this method with the code editor.
        /// </summary>
        private void InternalStartup()
        {
            this.Startup += new System.EventHandler(ThisDocument_Startup);
            this.Shutdown += new System.EventHandler(ThisDocument_Shutdown);
        }

        #endregion
    }
}

File: 2003/VB101SamplesAll/VB101SamplesVSTO/WorkingWithTheActionsPane/WorkingWithTheActionsPane/ThisDocument.vb

Public Class ThisDocument

    Dim uc As ProductLookupUserControl
    Dim b2 As Button


    Private Sub ThisDocument_Startup(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Startup
        ' Create a new instance of the user control
        uc = New ProductLookupUserControl

        ' Create a test button
        Dim b1 As Button
        b1 = New Button
        b1.Text = "Button 1"
        AddHandler b1.Click, AddressOf Me.b1_Click

        ' Create another test button
        b2 = New Button
        b2.Text = "Button 2"
        AddHandler b2.Click, AddressOf Me.b2_Click

        ' Add all three controls to the ActionsPane, show it, then
        ' add an event handler to fire if the orientation of the
        ' ActionsPane changes.
        Me.ActionsPane.Controls.AddRange(New Control() {uc, b1, b2})
        Me.ActionsPane.Show()
        AddHandler ActionsPane.OrientationChanged, AddressOf Me.ActionsPane_OrientationChanged
    End Sub

    Private Sub b1_Click(ByVal sender As Object, ByVal e As EventArgs)
        MessageBox.Show("You clicked button 1")
    End Sub

    Private Sub b2_Click(ByVal sender As Object, ByVal e As EventArgs)
        MessageBox.Show("You clicked button 2")
    End Sub


    Private Sub ActionsPane_OrientationChanged(ByVal sender As Object, ByVal e As EventArgs)
        If (Me.ActionsPane.Orientation = Orientation.Vertical) Then
            Me.ActionsPane.StackOrder = Microsoft.Office.Tools.StackStyle.FromTop
        Else
            Me.ActionsPane.StackOrder = Microsoft.Office.Tools.StackStyle.FromLeft
        End If
    End Sub

    Private Sub ThisDocument_Shutdown(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Shutdown

    End Sub

End Class
